invenio_files_rest/views.py

```diff
@@ -595,21 +582,22 @@ def put(self, bucket_id, key, content_length=None, content_md5=None):
         try:
             # TODO: Pass storage class to get_or_create
             obj = ObjectVersion.create(bucket, key)
-            obj.set_contents(uploaded_file, size=content_length)
+            obj.set_contents(uploaded_file)
             db.session.commit()
             # TODO: Fix response object to only include headers?
-            return {'json': {
-                'checksum': obj.file.checksum,
-                'size': obj.file.size,
-                'verisionId': str(obj.version_id),
-            }}
+            return {
+                'json': {
+                    'checksum': obj.file.checksum,
+                    'size': obj.file.size,
+                    'versionId': str(obj.version_id),
+                }
+            }
+        except FileSizeError as e:
```

Comment:
If we make ``FileSizeError`` a subclass of ``invenio_rest.errors:RESTException`` and wrap above block in a ``with db.session.begin_nested():``, then we can probably completely remove the try/except block (i.e. also the SQLAlchemyError, as it will properly rollback and return a 500 error.